alf/algorithms/rl_algorithm.py

```diff
@@ -132,11 +149,72 @@ def action_distribution_spec(self):
         """Return the action distribution spec for the action distributions."""
         return self._action_distribution_spec
 
+    @property
+    def exp_observers(self):
+        """Return experience observers."""
+        return self._exp_observers
+
+    @abstractmethod
+    def observe(self):
+        """An algorithm can decide what to do with experience."""
+        pass
+
+    def prepare_metrics(self, metrics=[]):
```

Comment:
set_metrics might be a better name. Also should comment that len(metrics) should be >= 2 as required by the code at L208